### PR TITLE
[Issue-2458][Refactor] Refactor 'streampark-flink-proxy' module using java language.

### LINKED ISSUE #2458: [Refactor][flink-proxy] Refactor flink-proxy module using java language.
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Refactor flink-proxy module using java language.

### Usage Scenario

N.A

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by RocMarshal**
## What changes were proposed in this pull request

Issue Number: close #2458 

## Brief change log

 Refactor 'streampark-flink-proxy' module using java language.

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

**Comment by RocMarshal** [comment]
Hi, @zhoulii  Could you help to review this one ? Thank you ~

File: streampark-flink/streampark-flink-proxy/src/main/java/org/apache/streampark/flink/proxy/ChildFirstClassLoader.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+ */
+public class ChildFirstClassLoader extends URLClassLoader {
+
+  private static final Pattern FLINK_PATTERN =
+      Pattern.compile("flink-(.*).jar", Pattern.CASE_INSENSITIVE | Pattern.DOTALL);
+
+  private static final String JAR_PROTOCOL = "jar";
+
+  private static final List<String> PARENT_FIRST_PATTERNS =
+      new ArrayList<String>() {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-proxy/src/main/java/org/apache/streampark/flink/proxy/ChildFirstClassLoader.java:47]: It is recommended to use: Arrays.as(...)

File: streampark-common/src/main/java/org/apache/streampark/common/util/Handler.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.streampark.common.util;
+
+/** A handler with declared exception thrown. */
+public interface Handler<T, R> {
```
Comment (wolfboys) [inline streampark-common/src/main/java/org/apache/streampark/common/util/Handler.java:21]: We don't need the new Fun interface, Just use java.util.function.Function

Comment (RocMarshal) [reply streampark-common/src/main/java/org/apache/streampark/common/util/Handler.java:21]: Hi, @wolfboys Thx for the comment.
However, If we use the` java.util.function.Function` without `Exception thrown declaration`, The `Utils.usingAdaptive` will be a lot of `try .... catch ...` clauses, which causes statements noisy.

Of course, I'd like to modify it based on your comments if we could ignored the negative points mentioned above.

**Pull request closed without merging**